### PR TITLE
Add environment.yml file for creating conda environment

### PR DESCRIPTION
**Pull request by soumyac1999**
Should the package versions also be specified in the the `environment.yml` file?

### CODE REVIEW

**Comment by lockshaw** [comment]
> Should the package versions also be specified in the the `environment.yml` file?

What's the difference between that and what you're currently doing?

**Comment by soumyac1999** [comment]
> > Should the package versions also be specified in the the `environment.yml` file?
> 
> What's the difference between that and what you're currently doing?

That is what I am currently doing. I am not entirely sure if some of the package versions are dependent on the machine config (probably not with these packages). I have only tried it on one machine.
Also, maintaining the file with specific version might be difficult (?).

**Comment by lockshaw** [comment]
> That is what I am currently doing. I am not entirely sure if some of the package versions are dependent on the machine config (probably not with these packages). I have only tried it on one machine. Also, maintaining the file with specific version might be difficult (?).

That seems doable--we can just bump the versions every so often, unless @eddy16112 or @jiazhihao think differently

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  - lerc=4.0.0=h27087fc_0
+  - libblas=3.9.0=16_linux64_openblas
+  - libcblas=3.9.0=16_linux64_openblas
+  - libdeflate=1.14=h166bdaf_0
+  - libffi=3.4.2=h7f98852_5
+  - libgcc-ng=12.1.0=h8d9b700_16
+  - libgfortran-ng=12.1.0=h69a702a_16
+  - libgfortran5=12.1.0=hdcd56e2_16
+  - libgomp=12.1.0=h8d9b700_16
+  - liblapack=3.9.0=16_linux64_openblas
```
Comment (eddy16112) [inline environment.yml:25]: Why do we need to install gcc and gfortran from conda?

File: environment.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  - libnsl=2.0.0=h7f98852_0
+  - libopenblas=0.3.21=pthreads_h78a6416_3
```
Comment (eddy16112) [inline environment.yml:27]: If we install openblas, we should not need cblas. Actually, I do not understand why we need openblas in FlexFlow.

File: environment.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  - openjpeg=2.5.0=h7d73246_1
+  - openssl=3.0.5=h166bdaf_2
+  - pillow=9.2.0=py310hbd86126_2
+  - pip=22.2.2=pyhd8ed1ab_0
+  - pthread-stubs=0.4=h36c2ea0_1001
+  - pybind11=2.10.0=py310hbf28c38_0
+  - pybind11-global=2.10.0=py310hbf28c38_0
+  - pycparser=2.21=pyhd3eb1b0_0
+  - python=3.10.6=ha86cf86_0_cpython
+  - python_abi=3.10=2_cp310
```
Comment (eddy16112) [inline environment.yml:47]: Do we need python 3.10?

File: environment.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  - readline=8.1.2=h0f457ee_0
+  - scipy=1.9.1=py310hdfbd76f_0
+  - setuptools=65.4.1=pyhd8ed1ab_0
```
Comment (eddy16112) [inline environment.yml:50]: Why do we need scipy?

**Comment by eddy16112** [comment]
I do not understand why we need so many packages? We used to only require packages listed here https://github.com/flexflow/FlexFlow/blob/master/requirements.txt

**Comment by soumyac1999** [comment]
This was a list of dependencies that were created when installing the packages in `requirements.txt` using conda. I obtained this using `conda list`. I can try removing dependencies of dependencies from the list. That should reduce the number of explicit dependencies.

**Pull request closed without merging**